### PR TITLE
Cpp98 removal

### PR DESCRIPTION
**Pull request by pepone**
This PR removes the C++98 mapping and the usage of ICE_CPP11_MAPPING. We have a single C++ mapping now.

There are more things to cleanup, like removing the usage of helper macros like ICE_UNCHECKED_CAST, but I let this for follow-up PRs.

### CODE REVIEW

File: cpp/msbuild/ice.cpp.props [hunk 1]
```diff
@@ -94,17 +99,17 @@
   <ItemDefinitionGroup Condition="'$(ICE_BIN_DIST)' != 'all'">
     <ClCompile>
       <AdditionalIncludeDirectories>$(IceHome)\cpp\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <AdditionalIncludeDirectories>$(IceHome)\cpp\include\generated\$(IceCppMapping)\$(Platform)\$(IceConfiguration);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(IceHome)\cpp\include\generated\cpp11\$(Platform)\$(Configuration);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (bernardnormier) [inline cpp/msbuild/ice.cpp.props:102]: Why cpp11?

Comment (bernardnormier) [reply cpp/msbuild/ice.cpp.props:102]: I guess it's related to the TODO: remove cpp11 in ice.test.props.

File: cpp/src/slice2cpp/Gen.cpp [hunk 2]
```diff
@@ -987,95 +978,6 @@ Slice::Gen::generate(const UnitPtr& p)
         Cpp11CompatibilityVisitor compatibilityVisitor(H, C, _dllExport);
         p->visit(&compatibilityVisitor, false);
     }
-
```
Comment (bernardnormier) [inline cpp/src/slice2cpp/Gen.cpp:990]: You should also remove all these C++98 visitors: ProxyDeclVisitor, DeclVisitor, etc. I am pretty sure they are independent of the Cpp11 visitors.

**Review by bernardnormier — COMMENTED** [review]
Looks good. There is only one CPP11 left:
```
cpp/doxygen/config-cpp11
2173:PREDEFINED             = ICE_CPP11_MAPPING
```

**Review by bernardnormier — APPROVED** [review]

**Review by externl — APPROVED** [review]
Looks good! We should wait for CI before merging.

**Pull request merged**